block-storage/src/main/scala/coop/rchain/blockstorage/dag/BlockDagKeyValueStorage.scala

```diff
@@ -62,6 +65,8 @@ final class BlockDagKeyValueStorage[F[_]: Concurrent: Log] private (
 
     def invalidBlocks: F[Set[BlockMetadata]] = invalidBlocksSet.pure[F]
 
+    override def lastFinalizedBlock: BlockHash = lastFinalizedBlockHash
```

Comment:
I got another question about this new concept on finalization. Is it possible the last finalized block could be several blocks in the new definitions instead of only one last finalized block in single parent mode ?

Reply:
LFB search area is constrained by blocks that are on intersection of main parent chains of tips holding more then 50% of the stake. So it always can be only one block at particular height, whether it's single or multi parent.